### ISSUE TITLE
Fix amip with diagnostic edmf longrun

### ISSUE THREAD

**Issue by szy21** [CliMA/ClimaCoupler.jl#641]
The amip target with diagnostic edmf longrun is currently failing in the most recent [build](https://buildkite.com/clima/climacoupler-longruns/builds/453#018de1db-cc9d-4c59-bcdb-130b0c99c440). We should fix it soon. Maybe try increasing hyperdiffusion and/or sponge.

**Comment by LenkaNovak** [comment]
@szy21 Here are some plots at the time of crash after three hours. The NaNs are coming from radiation (maybe related to the rrtmgp issue?) but there are other aspects we should address in this config to ensure the spin up is as smooth as possible. For example, the ocean albedo (atmos is now released for the more realistic version), air density is too low over topo (maybe this will get better with sim time though), initial land T is perhaps too warm, and roughness lengths over land and ocean should be adjusted to more realistic values. 


 
![debug_coupler](https://github.com/CliMA/ClimaCoupler.jl/assets/38083334/0e4a7e17-987d-4f8c-9271-af97c4caf3f0)

**Comment by LenkaNovak** [comment]
w fluctuations are pretty high around Antarctica too:
![debug_ClimaAtmosSimulation](https://github.com/CliMA/ClimaCoupler.jl/assets/38083334/40a31270-eeae-4b18-b7db-1121608a8998)

**Comment by szy21** [comment]
> @szy21 Here are some plots at the time of crash after three hours. The NaNs are coming from radiation (maybe related to the rrtmgp issue?) but there are other aspects we should address in this config to ensure the spin up is as smooth as possible. For example, the ocean albedo (atmos is now released for the more realistic version), air density is too low over topo (maybe this will get better with sim time though), initial land T is perhaps too warm, and roughness lengths over land and ocean should be adjusted to more realistic values.
> 
> ![debug_coupler](https://private-user-images.githubusercontent.com/38083334/308661311-0e4a7e17-987d-4f8c-9271-af97c4caf3f0.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDkxNDQ1NTgsIm5iZiI6MTcwOTE0NDI1OCwicGF0aCI6Ii8zODA4MzMzNC8zMDg2NjEzMTEtMGU0YTdlMTctOTg3ZC00ZjhjLTkyNzEtYWY5N2M0Y2FmM2YwLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDAyMjglMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwMjI4VDE4MTczOFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWFhMzk3N2Y5NGQ1ZThkODYzZTA4MDhkZDk2ZDRkOWNmZjMyMzBlN2FiNDRhZjRjZjc2Y2I5MGI3YTVlNWQwOGEmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.IjkAD0e_7sMHTWf8D_yUoVT0jNI2xDF4dpB10AXrRZc)

Oh, great, thank you! If it's NaNs in radiation we know how to fix it. It should be already fixed in the latest RRTMGP version.

**Comment by szy21** [comment]
But this is still with Float64 radiation, I'm a bit surprised it could have issues too. But let's see if the latest RRTMGP version can fix it.

**Comment by szy21** [comment]
Reopen it to track if we can change back `use_reference_state` to true

**Comment by LenkaNovak** [comment]
This now runs to completion with the new update to ClimaAtmos and ClimaLand (see [this build](https://buildkite.com/clima/climacoupler-longruns/builds/461#018df668-6754-4958-9b3a-b922bc47bb02)).

 
<img width="1141" alt="Screen Shot 2024-03-01 at 1 51 23 PM" src="https://github.com/CliMA/ClimaCoupler.jl/assets/38083334/2cee1bb5-1ca6-4281-bb9f-2b727bde3004">

Note the albedo hooks need still need to be added (see #660 ), but getting close.

**Comment by szy21** [comment]
Thanks! I actually want to change `use_reference_state` back to true. I can start a build after the albedo hook is added.

**Comment by szy21** [comment]
It's still breaking after the update: https://buildkite.com/clima/climacoupler-longruns/builds/465#018e0aba-da1e-4b67-846b-1efe3742af3e. The latest climaatmos release should include all the radiation fixes, we should check if radiation still has NaNs.

**Comment by szy21** [comment]
Here is the plot for the breaking simulation at hour 3. Radiation seems fine. It might break due to the large vertical velocity near the topography. I will try increasing hyperdiffusion.

![image](https://github.com/CliMA/ClimaCoupler.jl/assets/11598433/6e685526-8af2-4f63-a6b1-2c3ffc000947)
![image](https://github.com/CliMA/ClimaCoupler.jl/assets/11598433/4d9a132c-97b3-4e3e-ae6f-be26bcac2847)

**Comment by szy21** [comment]
Yes, everything I have been talking about is with `use_reference_state: true`. (`use_reference_state: false` works fine in the main.) Here is the [build](https://buildkite.com/clima/climacoupler-longruns/builds/465#018e0aba-da1e-4b67-846b-1efe3742af3e) that breaks after ~2 hours. This is using the main after #632 is merged, but I don't think any latest change should change the result?

I start to think that there might be something wrong with the combination of reference state, topography, and edmf. Given that `use_reference_state: false` works fine, maybe we don't need to worry too much about it for now, if there is nothing obviously wrong. I'll think more carefully about the reference state.

**Comment by szy21** [comment]
The error message shows that there is a negative number in sqrt in RRTMGP. I don't know if it's because the state is wrong or something inside RRTMGP. I'll look into it.

**Comment by LenkaNovak** [comment]
Sounds good, thanks! And in atmos this combination of reference state, topography, and edmf? Even if we use the slab ocean? 🤔

**Comment by szy21** [comment]
No, we test edmf and topography separately in atmos as I think it makes more sense to have land with topography. But if this problem persists I can go back to aqualanet for testing. I did test ref state + no topo + edmf in coupler and it seems to work. I may add that to longrun if it’s consistently more stable.

**Comment by szy21** [comment]
This seems to work now with a stronger sponge: https://buildkite.com/clima/climacoupler-longruns/builds/478#018e1c1b-f16e-47f7-b6b7-dbaa79d48de2. I just realized that we have been using a stronger sponge for topography runs in the atmos, but not for edmf runs. As the coupler jobs use the toml file in atmos I'm using the (weaker) one for edmf. So this might be it. I think it might be easier to just specify the coupler toml file.
